### PR TITLE
release: handle more registries for promotion

### PR DESCRIPTION
**Pull request by burmudar**
For cloud-ephemeral we want release also to be pushed to it's registry so that people launch ephemeral environments for releases.

This refactor the promotion script to handle handle additional registries. As a follow up a util func will be added to handle formating of the additional registries to ensure they're separated by a ` ` so that array splitting can happen.

Part of https://github.com/sourcegraph/sourcegraph/issues/61086
## Test plan
Tested locally
vars
```
export VERSION=6.6.666
export PUBLIC_REGISTRY='index.docker.io/sourcegraph-dev'
export ADDITIONAL_PROD_REGISTRY="us.gcr.io/sourcegraph-dev us.gcr.io/sourcegraph/cloud-ephemeral"
export INTERNAL_REGISTRY="us.gcr.io/sourcegraph/internal"
```
```
./tools/release/promote_images.sh gitserver
--- Copying gitserver from private registry to public registries
docker pull "us.gcr.io/sourcegraph/internal/gitserver:6.6.666"
docker tag "us.gcr.io/sourcegraph/internal/gitserver:6.6.666" "index.docker.io/sourcegraph-dev/gitserver:6.6.666"
docker push "index.docker.io/sourcegraph-dev/gitserver:6.6.666"
docker tag "us.gcr.io/sourcegraph/internal/gitserver:6.6.666" "us.gcr.io/sourcegraph-dev/gitserver:6.6.666"
docker push "us.gcr.io/sourcegraph-dev/gitserver:6.6.666"
docker tag "us.gcr.io/sourcegraph/internal/gitserver:6.6.666" "us.gcr.io/sourcegraph/cloud-ephemeral/gitserver:6.6.666"
docker push "us.gcr.io/sourcegraph/cloud-ephemeral/gitserver:6.6.666"
```
Annotation
```
## Release: image promotions

| Name | From | To |
|---|---|---|
| gitserver | `us.gcr.io/sourcegraph/internal/gitserver:6.6.666` |  `index.docker.io/sourcegraph-dev/gitserver:6.6.666` `us.gcr.io/sourcegraph-dev/gitserver:6.6.666` `us.gcr.io/sourcegraph/cloud-ephemeral/gitserver:6.6.666` |

```

### CODE REVIEW

**Review by BolajiOlajide — APPROVED** [review]

File: tools/release/promote_images.sh [hunk 1]
```diff
@@ -12,6 +12,16 @@ if [ "$#" -lt 1 ]; then
   exit 1
 fi
 
+# We're transitioning to GAR because of DockerHub new rate limiting affecting GCP
+# See https://github.com/sourcegraph/sourcegraph/issues/61696
+while read -r registry; do
```
Comment (burmudar) [inline tools/release/promote_images.sh:17]: hold up

**Review by jhchabran — CHANGES_REQUESTED** [review]
I see some remaining occurence of `ADDITIONAL_PROD_REGISTRY`: 

CleanShot 2024-04-30 at 16 46 11@2x

**Comment by burmudar** [comment]
> I see some remaining occurence of `ADDITIONAL_PROD_REGISTRY`:
> 
> CleanShot 2024-04-30 at 16 46 11@2x

Updated. The `push_all` script can already handle multple registries so just updated the env var to be consistent

**Comment by burmudar** [comment]
> Issue on https://github.com/sourcegraph/sourcegraph/pull/62269/files#diff-08b887c877bc36e59a90a823b4211912dd155bd0c083a033352c0cd94f443137R81 but @burmudar is on it.
> 
> Pre-approving as I'll afk for an hour and it's an easily testable fix.

Testing script
```bash
#!/usr/bin/env bash

ADDITIONAL_PROD_REGISTRIES="index.docker.io us.gcr.io/sourcegraph-dev"

prod_registries=()

if [ -n "${ADDITIONAL_PROD_REGISTRIES}" ]; then
  IFS=' ' read -r -a registries <<< "$ADDITIONAL_PROD_REGISTRIES"
  prod_registries+=("${registries[@]}")
fi

for r in "${prod_registries[@]}"; do
  echo "registry: $r"
done

```

which produces
```
bash test.sh
registry: index.docker.io
registry: us.gcr.io/sourcegraph-dev
```

which shows that the added bash handles splitting of strings into appropriate array values

**Review by Chickensoupwithrice — APPROVED** [review]

**Pull request merged**